src/index.ts

```diff
@@ -1,51 +1,68 @@
 import { compactDecrypt, importJWK, importPKCS8 } from 'jose'
-import {
-  Client,
-  ClientAuthMethod,
-  generators,
-  Issuer,
-  ResponseType,
-} from 'openid-client'
+import { Client, generators, Issuer } from 'openid-client'
 
+import {
+  API_VERSION,
+  DEFAULT_SCOPE,
+  DEFAULT_SGID_CODE_CHALLENGE_METHOD,
+  SGID_AUTH_METHOD,
+  SGID_SIGNING_ALG,
+  SGID_SUPPORTED_FLOWS,
+} from './constants'
 import { convertPkcs1ToPkcs8 } from './util'
 
-const SGID_SIGNING_ALG = 'RS256'
-const SGID_SUPPORTED_FLOWS: ResponseType[] = ['code']
-const SGID_AUTH_METHOD: ClientAuthMethod = 'client_secret_post'
+type ConstructorParams = {
+  clientId: string
+  clientSecret: string
+  privateKey: string
+  redirectUris?: string[]
+  hostname?: string
+}
+
+type AuthorizationUrlParams = {
+  state: string
+  scope?: string | string[]
+  nonce?: string | null
+  redirectUri?: string
+  codeChallenge: string
+}
+
+type AuthorizationUrlReturn = { url: string; nonce?: string }
+
+type CallbackParams = {
+  code: string
+  nonce?: string | null
+  redirectUri?: string
+  codeVerifier: string
+}
 
 export class SgidClient {
+  private apiVersion: number
   private privateKey: string
-
   private sgID: Client
 
   constructor({
     clientId,
     clientSecret,
     privateKey,
-    redirectUri,
+    redirectUris,
     hostname = 'https://api.id.gov.sg',
-    apiVersion = 1,
-  }: {
-    clientId: string
-    clientSecret: string
-    privateKey: string
-    redirectUri?: string
-    hostname?: string
-    apiVersion?: number
-  }) {
+  }: ConstructorParams) {
+    this.apiVersion = API_VERSION
+
     // TODO: Discover sgID issuer metadata via .well-known endpoint
     const { Client } = new Issuer({
       issuer: new URL(hostname).origin,
-      authorization_endpoint: `${hostname}/v${apiVersion}/oauth/authorize`,
-      token_endpoint: `${hostname}/v${apiVersion}/oauth/token`,
-      userinfo_endpoint: `${hostname}/v${apiVersion}/oauth/userinfo`,
+      authorization_endpoint: `${hostname}/v${this.apiVersion}/oauth/authorize`,
+      token_endpoint: `${hostname}/v${this.apiVersion}/oauth/token`,
+      userinfo_endpoint: `${hostname}/v${this.apiVersion}/oauth/userinfo`,
       jwks_uri: `${new URL(hostname).origin}/.well-known/jwks.json`,
```

Comment:
Updates due to our OpenID Discovery changes:
1. New Issuer will be appended with `/v2`.
2. That implies for our OAuth and OpenID endpoints, we don't have to manually insert `/v2` anymore.

The server-side changes can be found [here](https://github.com/datagovsg/sgid-server/pull/445).

Reply:
okay thank you! as we discussed in DMs:
- since we are not currently using the well known endpoints but instead manually constructing our endpoints, change 2 does not apply
- change 1 has still been made and `v2` is appended to the `issuer` field